python/gui/qgscolorbutton.sip

```diff
@@ -1,329 +1,404 @@
+ color will be used
+%End
 
-    /** Copies the current color to the clipboard
-     * @see pasteColor
-     */
     void copyColor();
+%Docstring
+ Copies the current color to the clipboard
+ \see pasteColor
+%End
 
-    /** Pastes a color from the clipboard to the color button. If clipboard does not contain a valid
-     * color or string representation of a color, then no change is applied.
-     * @see copyColor
-     */
     void pasteColor();
+%Docstring
+ Pastes a color from the clipboard to the color button. If clipboard does not contain a valid
+ color or string representation of a color, then no change is applied.
+ \see copyColor
+%End
 
-    /** Activates the color picker tool, which allows for sampling a color from anywhere on the screen
-     */
     void activatePicker();
+%Docstring
+ Activates the color picker tool, which allows for sampling a color from anywhere on the screen
+%End
 
-    /** Sets color to a totally transparent color.
-     * @note If the color button is not set to show an alpha channel in the color
-     * dialog (see setColorDialogOptions) then the color will not be changed.
-     */
     void setToNoColor();
+%Docstring
+ Sets color to a totally transparent color.
+.. note::
+
+   If the color button is not set to show an alpha channel in the color
+ dialog (see setColorDialogOptions) then the color will not be changed.
+ \see setToNull()
+%End
 
-    /** Sets color to the button's default color, if set.
-     * @see setDefaultColor
-     * @see defaultColor
-     */
     void setToDefaultColor();
+%Docstring
+ Sets color to the button's default color, if set.
+ \see setDefaultColor
+ \see defaultColor
+ \see setToNull()
+%End
 
-    /** Sets color to null.
-     * @see setToDefaultColor()
-     * @see setToNoColor()
-     * @note added in QGIS 2.16
-     */
     void setToNull();
+%Docstring
+ Sets color to null.
+ \see setToDefaultColor()
+ \see setToNoColor()
+.. versionadded:: 2.16
+%End
 
   signals:
 
-    /** Is emitted whenever a new color is set for the button. The color is always valid.
-     * In case the new color is the same no signal is emitted, to avoid infinite loops.
-     * @param color New color
-     */
     void colorChanged( const QColor &color );
+%Docstring
+ Is emitted whenever a new color is set for the button. The color is always valid.
+ In case the new color is the same no signal is emitted, to avoid infinite loops.
+ \param color New color
+%End
 
-    /** Emitted when the button is clicked, if the button's behavior is set to SignalOnly
-     * @param color button color
-     * @see setBehavior
-     * @see behavior
-     */
     void colorClicked( const QColor &color );
+%Docstring
+ Emitted when the button is clicked, if the button's behavior is set to SignalOnly
+ \param color button color
+ \see setBehavior
+ \see behavior
+%End
 
   protected:
 
-    void changeEvent( QEvent* e );
-    void showEvent( QShowEvent* e );
+    bool event( QEvent *e );
+%Docstring
+ :rtype: bool
```

Comment:
Note to self - @nyalldawson we need to avoid including rtype docstrings for overridden methods